keps/140-scenario-based-simulation/README.md

```diff
@@ -43,368 +64,539 @@ Therefore, this kep proposes to define the scenario **as CRD**. All clients, inc
 
-type ScenarioPluginsResults struct {
-  // Filter has each filter plugin’s result.
-  Filter            map[NodeName]map[PluginName]string
-  // Score has each score plugin’s score.
-  Score             map[NodeName]map[PluginName]ScenarioPluginsScoreResult
-}
+- Users can define Scenario and see how the simulated controllers work with a scenario.
+- Users can analyze and evaluate the controller with .status.Result.Timeline.
+	- All operations that happened while running the scenario will be recorded in Timeline and users can analyze Timeline by using our functions.
+- The result from the same Scenario won't be much changed run by run.
+	- Hopefully, if the Scenario and controllers are the same, the result should be similar. 
+
+#### The required configuration for users
+
+- add the comment directive to all simulated controller's codes and modify the code by our code generator.
+  - The simulator has the scheduler internally by default, and this scheduler has already been set up. 
+  - See [# How to stop the simulated controllers loop](#How-to-stop-the-simulated-controllers-loop).
+- define the functions for controllers to expect when controllers start to work and when controllers finish working.
+  - The simulator has some controllers internally by default, and they have already been set up. 
+  - See [# Expect when controllers start to work and when controllers finish working](#Expect-when-controllers-start-to-work-and-when-controllers-finish-working).
+
+#### Kubernetes controller
+
+Let's talk about the controllers in Kubernetes first.
+
+> In Kubernetes, controllers are control loops that watch the state of your cluster, then make or request changes where needed. Each controller tries to move the current cluster state closer to the desired state.
+https://kubernetes.io/docs/concepts/architecture/controller/
 
-type ScenarioPluginsScoreResult struct {
-  // RawScore has the score from Score method of Score plugins.
-  RawScore             int64
-  // NormalizedScore has the score calculated by NormalizeScore method of Score plugins.
-  NormalizedScore      int64
-  // FinalScore has score plugin’s final score calculated by normalizing with NormalizedScore and applied Score plugin weight. 
-  FinalScore           int64
+The Kubernetes controller has the desired state.
+For example, the ReplicaSet controller. It's working on managing the number of Pods to satisfy all replicaset's `.spec.replicas`.
+
+Some controllers in this world refer to something other than k8s resources like external metrics and real time to determine their ideal state. But, they cannot be used in our Scenario. In other words, **controllers that refer only to resources on k8s can be used in Scenario.**
+This is because the scenario controller cannot manage something other than k8s resources, and unexpected operations based on them may happen during the Scenario, which we want to avoid.
+
+The following controllers cannot be used in Scenario, for example:
```

Comment:
> This is because the scenario controller cannot manage something other than k8s resources, and unexpected operations based on them may happen during the Scenario, which we want to avoid.

Isn't the explanation above enough? What do you expect it?

Reply:
oh, ok. I think it's enough too. thanks.